File: Data Structures & Algorithms/subtree-of-a-binary-tree/submission-1.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */

class Solution {
public:
bool sametree(TreeNode* root1,TreeNode* root2 ){

    if(!root1 && !root2) return true;

    if(root1 && root2 && root1->val==root2->val)
    return sametree(root1->left,root2->left) && sametree(root1->right,root2->right);
    
    return false;
    }
    bool isSubtree(TreeNode* root, TreeNode* subRoot) {
        if(!subRoot) return true;

        if(!root && subRoot) return false;

        if(sametree(root,subRoot)) 
        return true;
        else 
        return isSubtree(root->left,subRoot)||isSubtree(root->right,subRoot);


        
    }
};

File: Data Structures & Algorithms/kth-smallest-integer-in-bst/submission-1.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        lst=[]
        # if root is None:
        #     return 
        # while root:
        def dfs(root): 
            if root is None:
                return           
            if root is not None:               
                lst.append(root.val)
            dfs(root.left)
            dfs(root.right)
        # lst.sort()
        dfs(root)
        lst.sort()
        print(lst)
        return lst[k-1]

        